nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-2-6-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/ConsumerLease.java

```diff
@@ -593,6 +609,9 @@ private void writeRecordData(final ProcessSession session, final List<ConsumerRe
                     try {
                         Record record;
                         while ((record = reader.nextRecord()) != null) {
```

Comment:
There's a special case that we need to consider here.
It's possible for a Kafka record to have a null value. In that case, it will produce no NiFi Records using the Record Reader. However, if using the Output Strategy of Wrapper, we should still output the wrapper. It may have a Key or Headers that are still of value.
So here, we should keep track of the fact that our Record Reader has produced a Record. When we end this `while` loop, if no NiFi Record was produced, we should output a Record where the value is `null` but the Key and Headers are still populated.

Reply:
Rather than tracking state, would it be reasonable to predetermine the decision based on `consumerRecord.value()` a few lines above?

```
final byte[] recordBytes = consumerRecord.value() == null ? new byte[0] : consumerRecord.value();
```

(If it is possible to emit no record even with non-null data, I guess this wouldn't work.)

Reply:
It's possible to have bytes that produce no record. For example, it could be JSON and contain `{}` or it could be a CSV message with just a header line and no actual data.

Reply:
It looks like it would be necessary to extract most of the logic inside the while loop into a helper method, as it would also be needed to process this null record.  Am I missing a simpler way to do this?